crates/state-aggregate/src/lib.rs

```diff
@@ -0,0 +1,85 @@
+use serde::{Deserialize, Serialize};
+
+use flat_serialize_macro::FlatSerializable;
+
+// Intermediate state kept in postgres.
```

Comment:
The `/crates` area is more intended to store postgres agnostic algorithm implementations.  The trans state, and the rest of the objects here should go in `extension\src\state_aggregate.rs` as it isn't really useful outside that context.